solidity/lockcontract/contracts/LockContract.sol

```diff
@@ -1,21 +1,65 @@
-pragma solidity ^0.4.17;
+pragma solidity ^0.4.23;
 
 contract LockContract {
-    string public message;
 
-    function LockContract(string initialMessage) public {
-        message = initialMessage;
+    struct Booking {
+        uint offerID;
+        uint256 checkIn;
+        uint256 checkOut;
+        address tenant;
     }
 
-    function setMessage(string newMessage) public {
-        message = newMessage;
+    struct Offer{
+        string price;   //Fixed Numbers currently not supported: https://github.com/ethereum/solidity/issues/409
+        string objectName;
+        string ownerName;
+        address owner;
+        uint[] bookingIndexes;
     }
 
-    function setBooked() public {
-        message = "Booked!";
+    Offer[] public offers;
+    Booking[] public bookings;
+
+    function insertOffer(string price, string objectName, string ownerName) public {
```

Comment:
In der Aufgabenbeschreibung steht, dass ein Angebot noch mehr Daten besitzt, die ich beim Erstellen angeben muss.

Reply:
Unabhängig davon, dass oben noch WIP steht: 
Auf welchen Punkt beziehst du dich genau in der Aufgabenbeschreibung und was fehlt dir hier

Reply:
Aufbau eines Angebots:
* Tür ID
* Zeitraum (Check-In/Check-Out) bzw für welchen Zeitraum ich das Zimmer vermieten möchte

Reply:
Die TürID sollte nicht vom Nutzer, sondern vom System gesetzt werden. __Begründung__ Woher soll der Nutzer wissen, welche IDs frei sind

Der Zeitraum sollte nicht bei Angebotserstellung vom Vermieter gesetzt werden. __Begründung__ Der Zeitraum wird vom Mieter bestimmt.

Reply:
Was Hyperledger betrifft, habe ich hierzu ein Issue aufgemacht: #26

Reply:
Nein ich möchte mein Zimmer nur für einen bestimmten Zeitraum vermieten, also muss ich den auch mit angeben können beim Erstellen. In der Aufgabenbeschreibung haben wir genau festgelegt welche Daten zu einem Angebot gehören. Die TürID muss auch der Vermieter angeben, da das System gar nicht wissen kann welche Tür mit dem Angebot verbunden sein soll.

Reply:
Ach TürID, sorry, da habe ich mich verlesen. Ja die fehlt noch.
Was den Zeitraum betrifft: Dann ist das Naming hier falsch, da CheckIn/Out bedeutet, dass ich dort meine Meinung beginne und beende. Du sprichst aber von einem befristetem Inserat, was nichts mit dem vom Mieter gewählten Zeitraum der Miete zu tun hat. Das ist aber so in der Aufgabenbeschreibung noch nicht berücksichtigt und sollte noch angepasst werden.